iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift

```diff
@@ -349,54 +362,86 @@ class SwipeableInputAccessoryView: View, AttachmentViewControllerDelegate, UIGes
         self.addSubview(self.previewView!)
 
         self.initialPreviewOrigin = self.previewView?.origin
+        self.currentSendPosition = nil
+
+        self.delegate.swipeableInputAccessoryDidBeginSwipe(self)
     }
 
-    private func handlePanChanged(withOffset offset: CGPoint, maxXOffset: CGFloat, maxYOffset: CGFloat) {
+    private func handlePanChanged(withOffset panOffset: CGPoint) {
         guard let initialPosition = self.initialPreviewOrigin else { return }
 
-        let offsetX = clamp(offset.x, -maxXOffset, maxXOffset)
-        let offsetY = clamp(offset.y, -maxYOffset, 0)
+        let offsetX = clamp(panOffset.x, -self.maxXOffset, self.maxXOffset)
+        let offsetY = clamp(panOffset.y, -self.maxYOffset, 0)
         self.previewView?.origin = initialPosition + CGPoint(x: offsetX, y: offsetY)
-    }
 
-    private func handlePanEnded(withProgress progress: CGFloat) {
-        let didFinish = progress > 0.5
+        guard let sendable = self.sendable else { return }
 
-        if didFinish {
-            self.selectionFeedback.impactOccurred()
-            self.previewView?.removeFromSuperview()
+        let newSendPosition = self.getSendPosition(forPanOffset: panOffset)
 
-            if let object = self.sendableObject {
-                if progress > 0.99 {
-                    self.delegate.swipeableInputAccessory(self, didConfirmReply: object)
-                } else {
-                    self.delegate.swipeableInputAccessory(self, didConfirm: object)
-                }
+        // Detect if the send position has changed. If so, let the delegate know so it can prepare
+        // for a send or cancel the current send.
+        if newSendPosition != self.currentSendPosition {
```

Comment:
We only update the delegate about preparing to send if the swipe position has changed. For example, you won't get two calls in a row for swipe position "middle".